modules/platform/nuxeo-platform-web-common/src/test/java/org/nuxeo/ecm/platform/web/idempotency/TestNuxeoIdempotentFilter.java

```diff
@@ -0,0 +1,406 @@
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.nuxeo.ecm.platform.web.common.idempotency.NuxeoIdempotentFilter;
+import org.nuxeo.runtime.kv.KeyValueService;
+import org.nuxeo.runtime.kv.KeyValueStoreProvider;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.RuntimeFeature;
+
+/**
+ * Checks idempotent requests management.
+ *
+ * @since 11.5
+ */
+@RunWith(FeaturesRunner.class)
+@Features(RuntimeFeature.class)
+@Deploy("org.nuxeo.runtime.kv")
+@Deploy("org.nuxeo.ecm.platform.web.common:OSGI-INF/idempotency-configuration.xml")
+public class TestNuxeoIdempotentFilter {
+
+    protected static final String KEY = "mykey";
+
+    protected static final String CONTENT = "test content";
+
+    protected static final Map<String, Collection<String>> RESPONSE_HEADERS = new LinkedHashMap<>();
+
+    static {
+        RESPONSE_HEADERS.put("Accept", List.of("text/html", "application/xhtml+xml", "*/*;q=0.8"));
+        RESPONSE_HEADERS.put("Connection", List.of("Keep-Alive"));
+        RESPONSE_HEADERS.put("Content-Encoding", List.of("gzip"));
+        RESPONSE_HEADERS.put("Content-Type", List.of("text/html; charset=utf-8"));
+        RESPONSE_HEADERS.put("Set-Cookie", List.of("sessionId=38afes7a8", "id=a3fWa; Max-Age=2592000"));
+        RESPONSE_HEADERS.put("Transfer-Encoding", List.of("chunked")); // should be filtered
+    }
+
+    protected static final Map<String, Collection<String>> KEY_RESPONSE_HEADERS = new LinkedHashMap<>();
+
+    static {
+        KEY_RESPONSE_HEADERS.put(NuxeoIdempotentFilter.HEADER_KEY, List.of(KEY));
+    }
+
+    protected static final Map<String, Collection<String>> FINAL_RESPONSE_HEADERS = new LinkedHashMap<>();
+
+    static {
+        FINAL_RESPONSE_HEADERS.putAll(RESPONSE_HEADERS);
+        FINAL_RESPONSE_HEADERS.putAll(KEY_RESPONSE_HEADERS);
+    }
+
+    protected static final Map<String, Collection<String>> FINAL_COPY_RESPONSE_HEADERS = new LinkedHashMap<>();
+
+    static {
+        FINAL_COPY_RESPONSE_HEADERS.putAll(FINAL_RESPONSE_HEADERS);
+        SKIPPED_HEADERS.forEach(FINAL_COPY_RESPONSE_HEADERS::remove);
```

Comment:
`FINAL_COPY_RESPONSE_HEADERS.keySet().removeAll(SKIPPED_HEADERS)` instead for clarity?